build_runner/lib/src/server/server.dart

```diff
@@ -651,7 +658,7 @@ shelf.Handler _logRequests(shelf.Handler innerHandler) {
           request.requestedUri, request.method, watch.elapsed);
       logFn(msg);
       return response;
-    }, onError: (Object error, StackTrace stackTrace) {
+    }).catchError((Object error, StackTrace stackTrace) {
```

Comment:
This change keeps correct the original stacktrace from the inner handler, as opposed to  creating a new one originating in _logResuests

Reply:
Can you elaborate on this? I'm not sure how this impacts the `StackTrace`.

Reply:
Just from my experiments, adding a `throw` statement inside the build runner server handler: 
- onError reports a stack trace originating in the `_logRequests` handler.
- catchError reports the original stack trace, starting in the server's handler.

I am not sure why the behavior is different though, I would expect it to be the same.

Reply:
I'm not able to to reproduce this. I tried adding a throw in `AssetHandler._handle` and the stack trace points there with either `onError:` or `.catchError`. Was this possibly related to other changes you had made? Can you still see a behavior difference?

Reply:
Let's go with https://github.com/dart-lang/build/pull/3149

If you can still repro a confusing stack trace after that one and #3148 are in we can revisit. I'm not sure I saw the _same_ confusing stack traces as you, but I'm pretty confident that the async/await pattern stands the best chance at giving us useful stack traces in the most situations on the VM since it has better handling for `await` than raw futures.

Reply:
sound good, thanks!

Reply:
done